cassiopeia/solver/SpectralGreedySolver.py

```diff
@@ -119,12 +123,14 @@ def perform_split(
         climbing method.
 
         Args:
-            samples: A list of samples, represented as integer indices
+            samples: A list of samples, represented by their string names
 
         Returns:
             A tuple of lists, representing the left and right partition groups
         """
-        mutation_frequencies = self.compute_mutation_frequencies(samples)
+        int_samples = list(map(lambda x: self.name_to_index[x], samples))
+
+        mutation_frequencies = self.compute_mutation_frequencies(int_samples)
```

Comment:
I noted this above, but I think that `compute_mutation_frequencies` should take in a list of sample names, not integer indices. You can map them to integer indices within the function itself if you'd like, but I think it's more usable if we pass in string names.